### PR TITLE
New Rule: not-match

### PR DESCRIPTION
**Pull request by TSMMark**
Solves #13

I also added an option `includePath: <boolean>` to support full-path matching (as opposed to just the file basename). I needed this because I realized that `AdBanner` was actually the name of a module, and so the path of the file was `whatever/AdBanner/index.tsx`. If you would prefer I split that into another PR I could do it that way. LMK

### CODE REVIEW

File: lib/match.js [hunk 1]
```diff
@@ -11,45 +12,21 @@ const meta = {
...
-};
-
 module.exports = {
   meta,
   create: (context) => ({
     Program: (node) => {
       const filename = context.getFilename();
-      const basename = path.basename(filename);
-      const [regex, regexStr] = getRegex(context.options[0], basename);
+      const includePath = !!((context.options[0] || {}).includePath);
```
Comment (dolsem) [inline lib/match.js:20]: You can just do
```suggestion
      const includePath = !!context.options[0]?.includePath;
```

File: lib/notMatch.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const path = require('path');
+const { getRegex } = require('./common/getRegex');
+
+const meta = {
+  type: 'layout',
+  docs: {
+    description: 'checks that filenames does not match a chosen pattern',
```
Comment (dolsem) [inline lib/notMatch.js:7]:
```suggestion
    description: 'checks that filenames do not match a chosen pattern',
```

**Review by dolsem — CHANGES_REQUESTED** [review]

File: lib/match.js [hunk 3]
```diff
@@ -11,45 +12,21 @@ const meta = {
...
-};
-
 module.exports = {
   meta,
   create: (context) => ({
     Program: (node) => {
       const filename = context.getFilename();
-      const basename = path.basename(filename);
-      const [regex, regexStr] = getRegex(context.options[0], basename);
+      const includePath = !!context.options[0]?.includePath;
```
Comment (TSMMark) [inline lib/match.js:20]: I don't think this project is set up to support optional prop chaining

```
SyntaxError: /Users/mark/code/eslint-plugin-filename-rules/lib/match.js: Unexpected token (20:47)
  18 |     Program: (node) => {
  19 |       const filename = context.getFilename();
> 20 |       const includePath = !!context.options[0]?.includePath;
     |                                                ^
  21 |       const name = includePath ? filename : path.basename(filename);
  22 |       const [regex, regexStr] = getRegex(context.options[0], name);
  23 |       if (!regex) return;
    at Parser.pp$5.raise (/Users/mark/code/eslint-plugin-filename-rules/node_modules/babylon/lib/index.js:4454:13)
```

**Comment by TSMMark** [comment]
resolving conflicts

**Comment by TSMMark** [comment]
@dolsem done

File: lib/match.js (hunk 3 above)
Comment (dolsem) [reply lib/match.js]: @TSMMark this project doesnt do any transpilation, so any syntax is only supported if your Node supports it. Are you using Node < v14? I can update package.json with a node version requirement. It's probably not worth supporting Node versions below 14, especially in the context of a development environment.

**Pull request merged**